Build failure: yubihsm-shell

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

`nix-build -A yubihsm-shell`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/308945487

### Relevant log output

```console
Using udevCheckHook
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/q0brrp8www7j1dwbsk9swhxifmk4gfiz-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
substituteStream() in derivation yubihsm-shell-2.7.0: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'lib/yubihsm.c')
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/share/doc/yubihsm-shell -DCMAKE_INSTALL_INFODIR=/nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/share/man -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/strip -DCMAKE_RANLIB=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ranlib -DCMAKE_AR=/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0
[0mCMake Deprecation Warning at CMakeLists.txt:17 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

[0m
-- The C compiler identification is GNU 14.3.0
-- The CXX compiler identification is GNU 14.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Certificate compression is enabled.
-- Performing Test HAVE_STACK_PROTECTOR_ALL
-- Performing Test HAVE_STACK_PROTECTOR_ALL - Success
-- -fstack-protector-all support detected
-- Performing Test HAVE_NO_IMPLICIT_FALLTHROUGH
-- Performing Test HAVE_NO_IMPLICIT_FALLTHROUGH - Success
-- not win32
-- Looking for memset_s
-- Looking for memset_s - not found
-- Looking for explicit_bzero
-- Looking for explicit_bzero - found
-- Found PkgConfig: /nix/store/k9s1w95rqfr14lb5xms1lwzrq1yiinzz-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for one of the modules 'libcrypto'
-- Checking for one of the modules 'libedit'
-- Checking for one of the modules 'libcurl'
-- Checking for one of the modules 'libusb-1.0'
[0mLIBCRYPTO_VERSION: 3.5.2[0m
-- Found ZLIB: /nix/store/6qi8skh85ci2k9gvl27nnh3kiy32qnsz-zlib-1.3.1/lib/libz.so (found version "1.3.1")
[33mCMake Warning at pkcs11/tests/CMakeLists.txt:165 (message):
  PKCS11TEST_PATH not defined, assuming that pkcs11test is in PATH

[0m
-- Using PKCS11_TESTER: /pkcs11test
-- Checking for one of the modules 'libpcsclite'
[0m/build/source/yubihsm-auth/cmdline.c[0m
[0m/build/source/yhwrap/cmdline.c[0m
[0mBuild summary:[0m
[0m[0m
[0m        Project name:     yubihsm-shell[0m
[0m        Version:          2.7.0[0m
[0m        Host type:        Linux[0m
[0m        Path prefix:      [0m
[0m        Compiler:         /nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/gcc[0m
[0m        Compiler ID:      GNU[0m
[0m        Compiler version: 14.3.0[0m
[0m        CMake version:    4.1.1[0m
[0m        CFLAGS:            -flto -Wno-missing-braces -Wno-missing-field-initializers[0m
[0m        CPPFLAGS:         [0m
[0m        Warnings:         [0m
[0m        Build type:       Release[0m
[0m[0m
[0m        Install prefix:    /nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0[0m
[0m        Install targets[0m
[0m             Libraries     /nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/lib[0m
[0m             Includes      /nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/include[0m
[0m             Binaries      /nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/bin[0m
[0m             Manuals       /nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/share/man[0m
[0m             Pkg-config    /nix/store/2g8fa4sqkyrynyancdnzy25m8wzc21hc-yubihsm-shell-2.7.0/share/pkgconfig[0m
-- Configuring done (3.1s)
-- Generating done (0.2s)
[33mCMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025

[0m
-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j16 SHELL=/nix/store/cl2gkgnh26mmpka81pc2g5bzjfrili92-bash-5.3p3/bin/bash
[  2%] [32mBuilding C object lib/tests/CMakeFiles/test_usb_url.dir/test_usb_url.c.o[0m
[  2%] [32mBuilding C object pkcs11/tests/CMakeFiles/aes_encrypt_test.dir/aes_encrypt_test.c.o[0m
[  2%] [32mBuilding C object lib/tests/CMakeFiles/test_util.dir/test_util.c.o[0m
[  2%] [32mBuilding C object lib/CMakeFiles/yubihsm_usb.dir/yubihsm_usb.c.o[0m
[  3%] [32mBuilding C object pkcs11/tests/CMakeFiles/rsa_enc_test.dir/rsa_enc_test.c.o[0m
[  3%] [32mBuilding C object pkcs11/tests/CMakeFiles/ecdh_derive_test.dir/ecdh_derive_test.c.o[0m
[  3%] [32mBuilding C object pkcs11/tests/CMakeFiles/ecdh_sp800_test.dir/ecdh_sp800_test.c.o[0m
[  5%] [32mBuilding C object pkcs11/tests/CMakeFiles/pkcs11_interfaces_test.dir/pkcs11_interfaces_test.c.o[0m
[  5%] [32mBuilding C object pkcs11/tests/CMakeFiles/pss_sign_test.dir/pss_sign_test.c.o[0m
[  2%] [32mBuilding C object lib/CMakeFiles/yubihsm.dir/__/aes_cmac/aes.c.o[0m
[  8%] [32mBuilding C object lib/CMakeFiles/yubihsm_http.dir/yubihsm_curl.c.o[0m
[  8%] [32mBuilding C object ykhsmauth/CMakeFiles/ykhsmauth_static.dir/error.c.o[0m
[  8%] [32mBuilding C object ykhsmauth/CMakeFiles/ykhsmauth.dir/error.c.o[0m
[  8%] [32mBuilding C object lib/CMakeFiles/yubihsm.dir/__/aes_cmac/aes_cmac.c.o[0m
[  8%] [32mBuilding C object examples/CMakeFiles/p11_generate_rsa.dir/p11_generate_rsa.c.o[0m
[  9%] [32mBuilding C object pkcs11/tests/CMakeFiles/asym_wrap_test.dir/asym_wrap_test.c.o[0m
[ 10%] [32mBuilding C object ykhsmauth/CMakeFiles/ykhsmauth.dir/ykhsmauth.c.o[0m
[ 10%] [32mBuilding C object ykhsmauth/CMakeFiles/ykhsmauth_static.dir/ykhsmauth.c.o[0m
[ 11%] [32mBuilding C object lib/tests/CMakeFiles/test_util.dir/__/lib_util.c.o[0m
[ 13%] [32mBuilding C object examples/CMakeFiles/p11_generate_rsa.dir/__/common/openssl-compat.c.o[0m
[ 13%] [32mBuilding C object lib/tests/CMakeFiles/test_usb_url.dir/__/lib_util.c.o[0m
[ 13%] [32mBuilding C object pkcs11/tests/CMakeFiles/pkcs11_interfaces_test.dir/common.c.o[0m
[ 14%] [32mBuilding C object pkcs11/tests/CMakeFiles/aes_encrypt_test.dir/common.c.o[0m
[ 14%] [32mBuilding C object lib/CMakeFiles/yubihsm_usb.dir/yubihsm_libusb.c.o[0m
[ 14%] [1m[32mLinking C executable p11_generate_rsa[0m
[ 14%] [32mBuilding C object pkcs11/tests/CMakeFiles/pss_sign_test.dir/common.c.o[0m
[ 15%] [32mBuilding C object lib/CMakeFiles/yubihsm.dir/__/common/hash.c.o[0m
[ 16%] [1m[32mLinking C static library libykhsmauth.a[0m
[ 17%] [32mBuilding C object lib/CMakeFiles/yubihsm_usb.dir/lib_util.c.o[0m
[ 18%] [1m[32mLinking C executable pkcs11_interfaces_test[0m
/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ar: CMakeFiles/ykhsmauth_static.dir/error.c.o: plugin needed to handle lto object
/nix/store/dmypp1h4ldn0vfk3fi6yfyf5yxp9yz0k-gcc-wrapper-14.3.0/bin/ranlib: libykhsmauth.a(error.c.o): plugin needed to handle lto object
[ 19%] [32mBuilding C object pkcs11/tests/CMakeFiles/rsa_enc_test.dir/common.c.o[0m
[ 19%] Built target ykhsmauth_static
[ 20%] [32mBuilding C object pkcs11/tests/CMakeFiles/ecdh_sp800_test.dir/common.c.o[0m
[ 20%] [1m[32mLinking C executable aes_encrypt_test[0m
[ 20%] [1m[32mLinking C executable test_util[0m
[ 20%] [32mBuilding C object pkcs11/tests/CMakeFiles/ecdh_derive_test.dir/common.c.o[0m
[ 20%] [1m[32mLinking C shared library libykhsmauth.so[0m
[ 20%] [32mBuilding C object lib/CMakeFiles/yubihsm_http.dir/lib_util.c.o[0m
[ 20%] [32mBuilding C object pkcs11/tests/CMakeFiles/asym_wrap_test.dir/common.c.o[0m
[ 20%] [32mBuilding C object lib/CMakeFiles/yubihsm.dir/__/common/pkcs5.c.o[0m
[ 21%] [1m[32mLinking C executable test_usb_url[0m
[ 22%] [1m[32mLinking C executable pss_sign_test[0m
[ 23%] [32mBuilding C object lib/CMakeFiles/yubihsm.dir/__/common/rand.c.o[0m
[ 24%] [1m[32mLinking C executable rsa_enc_test[0m
[ 24%] [1m[32mLinking C executable ecdh_sp800_test[0m
[ 24%] [32mBuilding C object lib/CMakeFiles/yubihsm.dir/__/common/ecdh.c.o[0m
[ 25%] [1m[32mLinking C shared library libyubihsm_usb.so[0m
[ 26%] [1m[32mLinking C executable ecdh_derive_test[0m
/nix/store/7h3qnwgvkw6z2r8lq4j5mks4l6r5x2cq-binutils-2.44/bin/ld: /nix/store/qhw0sp183mqd04x5jp75981kwya64npv-glibc-2.40-66/lib/Scrt1.o: in function `_start':
(.text+0x1b): undefined reference to `main'
collect2: error: ld returned 1 exit status
make[2]: *** [ykhsmauth/CMakeFiles/ykhsmauth.dir/build.make:117: ykhsmauth/libykhsmauth.so.2.7.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:961: ykhsmauth/CMakeFiles/ykhsmauth.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 27%] [1m[32mLinking C executable asym_wrap_test[0m
[ 28%] [32mBuilding C object lib/CMakeFiles/yubihsm.dir/error.c.o[0m
[ 29%] Built target pkcs11_interfaces_test
[ 29%] [1m[32mLinking C shared library libyubihsm_http.so[0m
[ 29%] Built target p11_generate_rsa
[ 29%] [32mBuilding C object lib/CMakeFiles/yubihsm.dir/yubihsm.c.o[0m
[ 30%] [32mBuilding C object lib/CMakeFiles/yubihsm.dir/data_compress.c.o[0m
[ 31%] [32mBuilding C object lib/CMakeFiles/yubihsm.dir/lib_util.c.o[0m
/nix/store/7h3qnwgvkw6z2r8lq4j5mks4l6r5x2cq-binutils-2.44/bin/ld: /nix/store/qhw0sp183mqd04x5jp75981kwya64npv-glibc-2.40-66/lib/Scrt1.o: in function `_start':
(.text+0x1b): undefined reference to `main'
collect2: error: ld returned 1 exit status
[ 31%] Built target test_util
[ 31%] Built target test_usb_url
make[2]: *** [lib/CMakeFiles/yubihsm_usb.dir/build.make:133: lib/libyubihsm_usb.so.2.7.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:513: lib/CMakeFiles/yubihsm_usb.dir/all] Error 2
[ 31%] Built target aes_encrypt_test
[ 31%] Built target pss_sign_test
/nix/store/7h3qnwgvkw6z2r8lq4j5mks4l6r5x2cq-binutils-2.44/bin/ld: /nix/store/qhw0sp183mqd04x5jp75981kwya64npv-glibc-2.40-66/lib/Scrt1.o: in function `_start':
(.text+0x1b): undefined reference to `main'
collect2: error: ld returned 1 exit status
make[2]: *** [lib/CMakeFiles/yubihsm_http.dir/build.make:117: lib/libyubihsm_http.so.2.7.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:545: lib/CMakeFiles/yubihsm_http.dir/all] Error 2
[ 31%] Built target rsa_enc_test
[ 31%] Built target ecdh_sp800_test
[ 31%] Built target ecdh_derive_test
[ 31%] Built target asym_wrap_test
[ 31%] [1m[32mLinking C shared library libyubihsm.so[0m
/nix/store/7h3qnwgvkw6z2r8lq4j5mks4l6r5x2cq-binutils-2.44/bin/ld: /nix/store/qhw0sp183mqd04x5jp75981kwya64npv-glibc-2.40-66/lib/Scrt1.o: in function `_start':
(.text+0x1b): undefined reference to `main'
collect2: error: ld returned 1 exit status
make[2]: *** [lib/CMakeFiles/yubihsm.dir/build.make:246: lib/libyubihsm.so.2.7.0] Error 1
make[1]: *** [CMakeFiles/Makefile2:481: lib/CMakeFiles/yubihsm.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.2, NixOS, 25.11 (Xantusia), 25.11.20251012.cf3f5c4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.94.0-pre20251013-dev_faa0b4f
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/anton/.config/nix/nix.conf:/nix/store/vril5d133zfqzzz2xf21fg3l529zkfv6-stylix-kde-config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/anton/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/anton/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/anton/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/s1h1qn58bjx2i08lbi55vngas9v3h78v-lix-2.94.0-pre20251013-dev_faa0b4f/share`
 - channels(root): `""`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`

### Notify maintainers

@MatthewCroughan 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/yu/yubihsm-shell/package.nix
+++ b/pkgs/by-name/yu/yubihsm-shell/package.nix
@@ -73,7 +73,10 @@ stdenv.mkDerivation rec {
   meta = with lib; {
     description = "Thin wrapper around libyubihsm providing both an interactive and command-line interface to a YubiHSM";
     homepage = "https://github.com/Yubico/yubihsm-shell";
-    maintainers = with maintainers; [ matthewcroughan ];
+    maintainers = with maintainers; [
+      matthewcroughan
+      numinit
+    ];
     license = licenses.asl20;
     platforms = platforms.all;
   };

--- a/pkgs/by-name/yu/yubihsm-shell/package.nix
+++ b/pkgs/by-name/yu/yubihsm-shell/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch,
   cmake,
   openssl,
   libusb1,
@@ -24,6 +25,13 @@ stdenv.mkDerivation rec {
     rev = version;
     hash = "sha256-ymGS35kjhNlFee3FEXF8n6Jm7NVaynjv+lpix6F75BQ=";
   };
+
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/Yubico/yubihsm-shell/pull/493.patch";
+      hash = "sha256-mM4ef1GV7BJT+EZ8B7+ejleTocglhxCWO/RKHZN69GE=";
+    })
+  ];
 
   postPatch = ''
     # Can't find libyubihsm at runtime because of dlopen() in C code

